pkg/resourcemanager/controller/networkpolicy/reconciler.go

```diff
@@ -331,10 +330,34 @@ func (r *Reconciler) reconcileEgressPolicy(
 		}}
 		networkPolicy.Spec.PodSelector = podLabelSelector
 		networkPolicy.Spec.PolicyTypes = []networkingv1.PolicyType{networkingv1.PolicyTypeEgress}
+	})
+}
+
+func (r *Reconciler) reconcilePolicy(
+	ctx context.Context,
+	log logr.Logger,
+	service *corev1.Service,
+	networkPolicyObjectMeta metav1.ObjectMeta,
+	podLabelSelector metav1.LabelSelector,
+	mutate func(*networkingv1.NetworkPolicy, metav1.LabelSelector),
+) error {
+	networkPolicy := &networkingv1.NetworkPolicy{ObjectMeta: networkPolicyObjectMeta}
+
+	effectivePodLabelSelector, mutated := shortenPodSelectorKeysIfTooLong(podLabelSelector)
+
+	_, err := controllerutils.GetAndCreateOrMergePatch(ctx, r.TargetClient, networkPolicy, func() error {
+		metav1.SetMetaDataLabel(&networkPolicy.ObjectMeta, resourcesv1alpha1.NetworkingServiceName, service.Name)
+		metav1.SetMetaDataLabel(&networkPolicy.ObjectMeta, resourcesv1alpha1.NetworkingServiceNamespace, service.Namespace)
 
+		mutate(networkPolicy, effectivePodLabelSelector)
 		return nil
 	}, controllerutils.SkipEmptyPatch{})
 
+	if mutated {
+		log.Info("Usual pod label selector contained at least one key exceeding 63 characters - it had to be mutated", "usualPodLabelSelector", podLabelSelector, "mutatedPodLabelSelector", effectivePodLabelSelector)
```

Comment:
How often are we going to see these messages? TBH, I'm not familiar enough with the netpol reconciler to understand if those messages will now pop up regularly or only once per netpol?

Reply:
These are logged/event-reported each time the controller tries to create the policy, i.e., at least at each restart of it (e.g., after VPA scaling), but also on relevant changes in the corresponding `Service`/`NetworkPolicy`/`Namespace`. See here to get to know what this controller reacts upon: https://github.com/gardener/gardener/blob/ce5e89272120d53adb41a0e78f1f1f5f7b8c2198/pkg/resourcemanager/controller/networkpolicy/add.go#L72-L88

Thinking about it, we might probably want to reduce the log level to `V(2)`, WDYT? On scale, this could probably lead to some good portion of "spam" in the logs. Nobody will too actively look into the logs anyways.